skyportal/facility_apis/sedm.py

```diff
@@ -0,0 +1,214 @@
+from . import FollowUpAPI
+from baselayer.app.env import load_env
+from datetime import datetime, timedelta
+import json
+import requests
+from requests_toolbelt.utils import dump
+
+env, cfg = load_env()
+
+
+def convert_request_to_sedm(request, method_value='new'):
+    photometry = sorted(request.obj.photometry, key=lambda p: p.mjd, reverse=True)
+    photometry_payload = {}
+
+    for p in photometry:
+        if (
+            p.filter.startswith('ztf')
+            and p.filter not in photometry_payload
+            and p.mag is not None
+        ):
+            # using filter[-1] as SEDM expects the bandpass name without "ZTF"
+            photometry_payload[p.filter[-1]] = {
```

Comment:
Looks like you may want to use DefaultDict here?

Reply:
Not quite, as we only want to include Keys that we actually have data for. Sometimes the dict can (and should) look like 

```
{'g': {...}}
```

whereas other times it is intentional that it look like

```
{'g': {...}, 'r': {...}, 'i': {...}}
```